### PR TITLE
Fix the fix for generic dashboards

### PR DESCRIPTION
**Pull request by ibacher**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [ ] My work includes tests or is validated by existing tests.

## Summary

Add fallback to `window.location` if `useLocation()` returns undefined. Should fix the fix from f19d297.

### CODE REVIEW

File: packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ export interface DashboardExtensionProps {
 }
 
 export const DashboardExtension = ({ title, basePath }: DashboardExtensionProps) => {
-  const location = useLocation<Location>();
+  const location = useLocation<Location>() || window.location;
```
Comment (ZacButko) [inline packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx:13]: Do you want to throw in `?? {pathname: ""}` just to be super extra safe?

**Review by brandones — APPROVED** [review]

**Review by ZacButko — APPROVED** [review]
Do we know why this fixes it? Is there some time when useLocation is unavailable? Or what is causing this to fail?

File: packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx (hunk 1 above)
Comment (ZacButko) [reply packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx:13]: Or just put optional chaining everywhere

**Comment by ibacher** [comment]
> Do we know why this fixes it? Is there some time when useLocation is unavailable? 

I haven't really tracked down the circumstances, but the error was that it's undefined... I'd suspect it's possible that one of the dashboards get mounted in a React tree that doesn't have a React-Router as a DOM. `window.location` should always be defined, though or else the browser is broken...

**Pull request merged**